syncapi/sync/request.go

```diff
@@ -69,23 +60,20 @@ func newSyncRequest(req *http.Request, device userapi.Device, syncDB storage.Dat
 				util.GetLogger(req.Context()).WithError(err).Error("gomatrixserverlib.SplitID failed")
 				return nil, err
 			}
-			f, err := syncDB.GetFilter(req.Context(), localpart, filterQuery)
-			if err == nil {
-				timelineLimit = f.Room.Timeline.Limit
+			if f, err := syncDB.GetFilter(req.Context(), localpart, filterQuery); err == nil {
+				filter = *f
+			} else {
+				panic(err)
```

Comment:
We shouldn't panic, just fail the request?

Reply:
You're right, that was from testing. It's fixed now.